tests/testthat/test-sample.R

```diff
@@ -146,3 +146,63 @@ test_that("need a direct sample function in order to start sampling", {
     mcstate_sample(model2, sampler, 10),
     "'initial' must be provided with this model")
 })
+
+
+test_that("can continue chains", {
+  model <- ex_simple_gamma1()
+  sampler <- mcstate_sampler_random_walk(vcv = diag(1) * 0.01)
+
+  set.seed(1)
+  res1 <- mcstate_sample(model, sampler, 100, 1, n_chains = 3)
+
+  set.seed(1)
+  res2a <- mcstate_sample(model, sampler, 50, 1, n_chains = 3,
+                          restartable = TRUE)
+  res2b <- mcstate_sample_continue(res2a, 50)
+
+  expect_equal(res2b, res1)
+})
+
+
+test_that("can continue continuable chains", {
+  model <- ex_simple_gamma1()
+  sampler <- mcstate_sampler_random_walk(vcv = diag(1) * 0.01)
+
+  set.seed(1)
+  res1 <- mcstate_sample(model, sampler, 30, 1, n_chains = 3,
+                         restartable = TRUE)
+
+  set.seed(1)
+  res2a <- mcstate_sample(model, sampler, 10, 1, n_chains = 3,
+                          restartable = TRUE)
+  res2b <- mcstate_sample_continue(res2a, 10, restartable = TRUE)
+  res2c <- mcstate_sample_continue(res2b, 10, restartable = TRUE)
+
+  expect_equal(res2c, res1)
+})
```

Comment:
So is the difference between these two tests just the extra res2c? Which we would not be able to do had we not included restartable=TRUE for res2b

Reply:
yeah, it tests that we can restart a chain that was restarted; so both that we get the same answer as the combo chain (`res1`) and that the restart information is copied over